### PR TITLE
Add ScipyDistribution

### PR DESCRIPTION
**Pull request by kawindme**
Add the ScipyDistribution class that can be used to easily turn scipy distributions into virocon distributions.

### CODE REVIEW

File: virocon/distributions.py [hunk 1]
```diff
@@ -1494,3 +1495,143 @@ def _fit_mle(self, sample):
...
+        # Returns
+            A list of distribution parameter strings.
+        """
+        if isinstance(distribution, str):
+            distribution = getattr(sts, distribution)
+        if distribution.shapes:
+            parameters = [name.strip() for name in distribution.shapes.split(",")]
+        else:
+            parameters = []
+        # assert distribution is continous
```
Comment (ahaselsteiner) [inline virocon/distributions.py:1567]: How does the next line ( parameters += ["loc", "scale"] ) assert that the distribution is continuous? Or is the comment wrong?

File: virocon/distributions.py [hunk 2]
```diff
@@ -1494,3 +1495,143 @@ def _fit_mle(self, sample):
...
+        for i, arg in enumerate(args):
+            if arg is not None:
+                args_with_default[i] = arg
+
+        for key, arg in kwargs.items():
+            try:
+                idx = self._param_names.index(key)
+            except ValueError:
+                # unknown parameter name
+                assert False
```
Comment (ahaselsteiner) [inline virocon/distributions.py:1594]: What is the purpose of "assert False"? I know 'assert' only from tests.

File: virocon/distributions.py [hunk 3]
```diff
@@ -1494,3 +1495,143 @@ def _fit_mle(self, sample):
...
+        scipy_par = self._get_scipy_parameters(*args, **kwargs)
+        return self.scipy_dist.pdf(x, *scipy_par)
+
+    def draw_sample(self, n, *args, random_state=None, **kwargs):
+        scipy_par = self._get_scipy_parameters(*args, **kwargs)
+        rvs_size = self._get_rvs_size(n, scipy_par)
+        return self.scipy_dist.rvs(*scipy_par, size=rvs_size, random_state=random_state)
+
+    def _fit_mle(self, sample):
+        p0 = [v for k, v in self.parameters.items() if k != "loc" and k != "scale"]
```
Comment (ahaselsteiner) [inline virocon/distributions.py:1617]: Could you add a comment to explain this line?

I guess it produces a list of all distribution parameters which are not named "loc" and "scale"?

So the result for a 4-parameter distribution that has loc and scale parameters might be [0.0 1.0] (if 0 and 1 are the default values? This is a bit of a guess, I don't understand it.)

File: virocon/distributions.py [hunk 4]
```diff
@@ -1494,3 +1495,143 @@ def _fit_mle(self, sample):
...
+        scale0 = self.parameters.get("scale", 1)
+
+        fparams = {}
+        for par_name in self._param_names:
+            val = getattr(self, f"f_{par_name}")
+            if val is not None:
+                fparams[f"f{par_name}"] = val
+
+        if len(fparams) == len(self.parameters):
+            return  # nothing to do
```
Comment (ahaselsteiner) [inline virocon/distributions.py:1628]: Should we raise a Warning or Error in that case? 

It's probably a mistake by the user if they call _fit_mle() but have fixed all parameters.

**Review by ahaselsteiner — COMMENTED** [review]
Great feature! Well-documented and in typical Kai-style very well-implemented. 

I added a couple of minor comments. In the documentation, I directly edited some typos and made suggestions for different wording (how we refer to the two approaches).

Feel free to merge.

File: manual_tests/manual_test_ScipyDistribution.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+try:
+    gengamma_by_dist = GeneralizedGammaDistributionByDist(floc=0)
+except TypeError:
+    pass
+
+gengamma_by_name = GeneralizedGammaDistributionByName(f_loc=0)
+gengamma_by_dist = GeneralizedGammaDistributionByDist(f_loc=0)
+
+
+# Done
```
[the remark this answers is not in the document]
Comment (kawindme) [reply manual_tests/manual_test_ScipyDistribution.py:51]: Removed

File: virocon/distributions.py (hunk 1 above)
Comment (kawindme) [reply virocon/distributions.py:1558]: Removed

File: virocon/distributions.py (hunk 2 above)
Comment (kawindme) [reply virocon/distributions.py:1585]: Removed

File: virocon/distributions.py (hunk 3 above)
Comment (kawindme) [reply virocon/distributions.py:1608]: Done.

**Pull request merged**